core/src/main/java/io/confluent/rest/SslFactoryPemHelper.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest;
+
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.security.ssl.DefaultSslEngineFactory;
+
+import java.security.KeyStore;
+import java.util.Objects;
+
+/**
+ * This class provides util methods to handle Pem type keystore. The primary aim
+ * is to encapsulate the DefaultSslEngineFactory invocation in a separate class.
+ * Based on the provider type, an intermediate keystore of appropriate type will be
+ * created.
+ */
+public class SslFactoryPemHelper {
```

Comment:
this could be a static inner class in SslFactory

Reply:
@trnguyencflt 
Wanted to avoid any references to DefaultSslEngineFactory in SslFactory. All code prone to modification has been encapsulated in the helper class.